lib/find-file.js

```diff
@@ -53,18 +53,14 @@ function findFile(opt) {
       // find path to the root, stopping at cwd, package.json or bower.json
       var pathToRoot = [];
       var isWorking;
-      var depthCount = 0;
       do {
         pathToRoot.push(absoluteStart);
-        if (!testNotPackage(absoluteStart)) {
-          depthCount += 1;
-        }
-        isWorking = testWithinLimit(absoluteStart) && depthCount < options.moduleDepth;
+        isWorking = testWithinLimit(absoluteStart) && testNotPackage(absoluteStart);
         absoluteStart = path.resolve(absoluteStart, '..');
       } while (isWorking);
 
       // start a queue with the path to the root
-      var queue = pathToRoot.concat();
+      var queue = pathToRoot.concat(!!options.root && pathToRoot.indexOf(options.root) === -1 ? limit : []);
```

Comment:
I think the defensive programming is a good call. I was assuming the `limit` would not be in the list but I reread the while loop and I think it may well be.

However, the logic is getting a little long in which case the precedent is to shift to a `var`.

* Place an additional line `var appendLimit = ..` between this line and the comment and use it to simplify the ternary. Feel free to choose a variable name you like.

Regarding the logic:

* `!!options.root` : We already have an `options.root` elsewhere so you will need `options.includeRoot` or something. They need to be independent.

* `pathToRoot.indexOf(options.root)` : This is more correctly `pathToRoot.indexOf(limit)`. The `limit` is actually `options.root || process.cwd()`.

**aside:** In the docs `process.cwd()` is supposed to be the default value of `options.root`. However that is only so in this function. It is probably better set as a default in `index.js` when we are doing `lodash.default()` on all the options. So then `options.root` would be validated up front and we could remove the `limit` and its mutations.